### ISSUE TITLE
Information lost between modules

### ISSUE THREAD

**Issue by thoradam** [facebook/flow#283]
Working example:

`ModuleA.js`
```javascript
/** @flow */
function get(): Promise<number> {
    return new Promise(function (res) {
        res(7);
    });
}

get().then(function (n) {
    console.log(n.length);
});
```

Flow output:
```
ModuleA.js:10:17,24: property length
Property not found in
lib/core.js:59:1,74:1: Number
```

In the example above Flow detects the error of calling `then` on a `number`. However if you export `get` and call it in the same way from another module Flow does not detect the error:

`ModuleA.js`
```javascript
/** @flow */
function get(): Promise<number> {
  return new Promise(function (res) {
    res(7);
  });
}
 
module.exports = get;
```

`ModuleB.js`
```javascript
/** @flow */
var get = require('./ModuleA.js');
 
get().then(function(n) {
  console.log(n.length);
});
```

Flow output:
```
No errors!
```

**Comment by samwgoldman** [comment]
I just tested this against a recent commit on `master` and I now see the expected error.

**Comment by samwgoldman** [comment]
This has been fixed in 0.13. Please reopen if you continue to have this issue after upgrading.
